From 46d2c2f8fecad85baeed9fe211f5285820220442 Mon Sep 17 00:00:00 2001
From: James Cowgill <james410@cowgill.org.uk>
Date: Mon, 5 Jan 2015 15:16:58 +0000
Subject: [PATCH] Ignore differences between mips machine identifiers

Little endian binaries actually use EM_MIPS so you can't tell the endianness
from the elf machine id. Also, the EM_MIPS_RS3_LE machine is dead anyway (the
kernel will not load binaries containing it).

Signed-off-by: James Cowgill <james410@cowgill.org.uk>
Upstream-Status: Pending [from debian]
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

---
 backends/mips_init.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/backends/mips_init.c b/backends/mips_init.c
index 975c04e..8482e7f 100644
--- a/backends/mips_init.c
+++ b/backends/mips_init.c
@@ -45,11 +45,7 @@ mips_init (Elf *elf __attribute__ ((unused)),
     return NULL;
 
   /* We handle it.  */
-  if (machine == EM_MIPS)
-    eh->name = "MIPS R3000 big-endian";
-  else if (machine == EM_MIPS_RS3_LE)
-    eh->name = "MIPS R3000 little-endian";
-
+  eh->name = "MIPS";
   mips_init_reloc (eh);
   HOOK (eh, reloc_simple_type);
   HOOK (eh, return_value_location);
